backend/inspirehep/disambiguation/cli.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import click
+from elasticsearch_dsl import Q
+from inspire_utils.record import get_values_for_schema
+
+from inspirehep.records.api import InspireRecord
+from inspirehep.search.api import AuthorsSearch, LiteratureSearch
+
+
+@click.group()
+def disambiguation():
+    """Commands for disambiguation"""
+
+
+@disambiguation.command()
+def clean_stub_authors():
+    stub_authors_query = Q("term", stub=True)
+    stub_authors = AuthorsSearch().query(stub_authors_query).execute()
+    for author in stub_authors:
+        author_data = author.to_dict()
+        author_bai = get_values_for_schema(author_data["ids"], "INSPIRE BAI")[0]
+        author_papers_query = Q(
+            "nested",
+            path="authors",
+            query=Q("match", authors__ids__value__raw=author_bai),
+        )
+        author_papers = LiteratureSearch().query(author_papers_query).execute()
```

Comment:
ES might be inconsistent with the DB, and here you're changing the DB based on ES data, that's dangerous. Two things might happen:
1. the author record is not actually `stub` any longer, so you shouldn't delete it even if it has no papers
2. the author record has papers associated, but ES doesn't know yet.

I think the best course of action would be to use the DB only for this. That might require some work if you want it to be efficient (adding indices/auxiliary tables). Alternatively, you could get `stub` record IDs from ES, use `get_records_by_pids` to get the DB records and confirm they're still stub and get the BAI, then use a query against the `RecordsAuthors` table to check if there are no associated papers (that probably requires adding a new index on that table). I believe you also need a higher transaction isolation level than the standard `read committed` (I would guess `serializable`), because you're removing author records based on a predicate (are there associated papers), which the standard optimistic concurrency approach using a version counter on the author record doesn't protect.